torch/onnx/_internal/fx/onnxfunction_dispatcher.py

```diff
@@ -125,30 +115,64 @@ def dispatch(
     def _find_the_perfect_or_nearest_match_onnxfunction(
         self,
         node: torch.fx.Node,
-        aten_name: str,
-        function_overloads: Set[
-            Union["onnxscript.OnnxFunction", "onnxscript.TracedOnnxFunction"]
+        domain: str,
+        op_name: str,
+        overload: Optional[str],
+        all_functions: Set[registration.SymbolicFunction],
+        custom_functions: Optional[Set[registration.SymbolicFunction]],
+        onnx_args: Sequence[
+            Optional[Union[fx_type_utils.TensorLike, str, int, float, bool, list]]
         ],
-        onnx_args: Sequence[Optional[Union[_TensorLike, str, int, float, bool, list]]],
         onnx_kwargs: Dict[str, fx_type_utils.Argument],
         diagnostic_context: diagnostics.DiagnosticContext,
     ):
-        """Find the perfect/nearest matched OnnxFunction for the given FX node, arguments, and keyword arguments."""
-        overload_match_ranking: Dict[
-            Union[onnxscript.OnnxFunction, onnxscript.TracedOnnxFunction], int
-        ] = {}
+        """Find the perfect/nearest matched OnnxFunction for the given FX node, arguments, and keyword arguments.
+
+        Args:
+            node: The TorchFX node to dispatch the function for.
+            domain: The domain of the ONNX node.
+            op_name: The name of the ONNX node.
```

Comment:
For my understanding, what is the domain here? Why is it needed? I thought the name is in the node, or are they different?

Reply:
Also overload?

Reply:
So the key in registry is actually aten::add.Tensor, not aten::add. So we need torch.ops.<domain>.<op_name>.<overload>.

Reply:
Should the node not have everything though?

Reply:
This is a "dispatched aten name". In `get_aten_name`, we have some processing on the original node.name.